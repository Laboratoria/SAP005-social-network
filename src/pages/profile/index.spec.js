import { Profile } from './index.js';
import * as services from '../../services/index.js';

describe('Profile', () => {
  beforeEach(() => {
    services.delUser = jest.fn(() => Promise.resolve(true));
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should be a function', () => {
    expect(typeof Profile).toBe('function');
  });
  it('shoul return profilepag', () => {
    expect(Profile()).toMatchSnapshot();
  });
  it('should delete user', () => {
    const profileTest = Profile();
    const delBtn = profileTest.querySelector('#deleteuser');
    delBtn.dispatchEvent(new Event('click'));
    expect(services.delUser).toHaveBeenCalled();
  });
  it('should logout user', () => {
    const profileTest = Profile();
    const outBtn = profileTest.querySelector('#logOut');
    outBtn.dispatchEvent(new Event('click'));
    expect(services.logOut).toHaveBeenCalled();
  });
});
