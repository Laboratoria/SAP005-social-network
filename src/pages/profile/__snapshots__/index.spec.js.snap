// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile shoul return profilepag 1`] = `
<div
  class="div-profile"
>
  
  
  <figure
    class="logo"
  >
    
  
    <img
      alt="Logo Runners"
      id="logo"
      src="./assets/logo_runners.png"
    />
    
  
  </figure>
  
  
  <form>
    
  
    <h3
      class="error"
      id="msgError"
    />
    
  
    <div
      class="profile-header"
    >
      
      
      <img
        alt="Imagem do Usuario"
        id="photo"
        src="https://github.com/elis-ctrl.png"
      />
      
      
      <p
        class="text"
        id="name"
      >
        Teste
      </p>
      
      
      <p
        class="text"
        id="email"
      >
        teste@teste1.com
      </p>
      
      
    </div>
    
      
    <div>
      
      
      
      <div
        class="profile-btn"
      >
        
      
        <button
          id="deleteuser"
        >
          Deletar Perfil
        </button>
        
      
        <button
          id="cancelbtn"
        >
          Cancelar
        </button>
        
      
      </div>
      
      
    </div>
  </form>
</div>
`;
